messagebus/src/main/java/com/yahoo/messagebus/network/rpc/RPCNetwork.java

```diff
@@ -301,15 +300,34 @@ public boolean destroy() {
         return false;
     }
 
+    private static Version deriveSupportedProtocolVersion() {
+        // This is a very leaky abstraction, but since MessageBus only exchanges versions
+        // (and not a set of supported protocols), we have to do this workaround.
+        // Disallow-version MUST be lower than that used as a protocol lower bound in
+        // DocumentProtocol.java and the exact same as that used in C++ for the same purposes.
+        // ... Or else!
+        // TODO remove this glorious hack once protobuf protocol is enabled by default
```

Comment:
🙈

Reply:
👀 💦

Reply:
Will this usage of `protobuf` be sufficient to never need the "version negotiation" again? I'm not super-familiar with this glorious new technology. I mean, where does _it_ check protocol versions, or whatever it does.

Reply:
Also, enabling the Protobuf protocol by default would just change the reported version to always be one that is >= the version that triggers both sides to use the correct codecs. The same dance could then be done again later if we wanted to add a new protocol with a version > the Protobuf protocol version.